int64[] error_box
string error_msg